pkg/server/policy.go

```diff
@@ -107,7 +107,7 @@ func (f *forbidden) Explanation() string {
 }
 
 func (p *RequestingAnnotatedRolePolicy) IsAllowedAssumeRole(ctx context.Context, role, podIP string) (Decision, error) {
-	pod, err := p.pods.GetPodByIP(ctx, podIP)
+	pod, err := p.pods.GetPodByIP(podIP)
```

Comment:
This is a point where I'm curious if we should validate the client IP (kiam client making the request) and the HostIp in the [PodStatus](https://godoc.org/k8s.io/api/core/v1#PodStatus) are a match? This would prevent being able to use a compromised host to provide IAM credentials for pods that could not be scheduled on the host. Probably a future enhancement but wanted to pose the question.

Reply:
It's only the agent that directly receives requests from clients, after which point the ip address forwarded by the agent is used to identify the pod. Are you suggesting the agent also pass its host ip to the server in this api call?

I suppose it does make sense to check that as a pod shouldn't hit another host's agent. @pingles would supporting this be another major bump or just an extension of the current request struct with graceful fallback?

Reply:
@moofish32 I like the suggestion as an additional security enhancement (I assume this means check that the pod's host matches the agent's client IP?)

If so- could you open a new issue to track it please? I'm not sure if we'd commit to adding for v3 but I don't think it'd change the API as-is (the Pod IP is passed to `GetPodCredentials` which would allow for additional policy to check against the `PodStatus`).

Reply:
@sambooo -- we have both actually. The agent passes the IP of the pod and [GRPC](https://godoc.org/google.golang.org/grpc/peer) enables us to track the agent IP. The issue is tracked as #105